aws/resource_aws_appconfig_environment.go

```diff
@@ -0,0 +1,240 @@
+	}
+
+	env, err := conn.CreateEnvironment(input)
+	if err != nil {
+		return fmt.Errorf("Creating AppConfig Environment failed: %s", err)
+	}
+	log.Printf("[DEBUG] AppConfig Environment created: %s", env)
+
+	d.SetId(aws.StringValue(env.Id))
+
+	return resourceAwsAppconfigEnvironmentRead(d, meta)
+}
+
+func resourceAwsAppconfigEnvironmentRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).appconfigconn
+	ignoreTagsConfig := meta.(*AWSClient).IgnoreTagsConfig
+
+	input := &appconfig.GetEnvironmentInput{
+		ApplicationId: aws.String(d.Get("application_id").(string)),
+		EnvironmentId: aws.String(d.Id()),
+	}
+
+	output, err := conn.GetEnvironment(input)
+
+	if isAWSErr(err, appconfig.ErrCodeResourceNotFoundException, "") {
+		log.Printf("[WARN] Appconfig Environment (%s) not found, removing from state", d.Id())
+		d.SetId("")
+		return nil
+	}
+
+	if err != nil {
+		return fmt.Errorf("error getting AppConfig Environment (%s): %s", d.Id(), err)
+	}
+
+	if output == nil {
+		return fmt.Errorf("error getting AppConfig Environment (%s): empty response", d.Id())
+	}
+
+	appARN := arn.ARN{
+		AccountID: meta.(*AWSClient).accountid,
+		Partition: meta.(*AWSClient).partition,
+		Region:    meta.(*AWSClient).region,
+		Resource:  fmt.Sprintf("application/%s/environment/%s", aws.StringValue(output.ApplicationId), aws.StringValue(output.Id)),
+		Service:   "appconfig",
+	}.String()
+
+	d.Set("arn", appARN)
+	d.Set("name", output.Name)
+	d.Set("description", output.Description)
+	d.Set("monitor", convertMonitorsToMap(output.Monitors))
+
+	tags, err := keyvaluetags.AppconfigListTags(conn, appARN)
+	if err != nil {
+		return fmt.Errorf("error getting tags for AppConfig Environment (%s): %s", d.Id(), err)
+	}
+
+	if err := d.Set("tags", tags.IgnoreAws().IgnoreConfig(ignoreTagsConfig).Map()); err != nil {
+		return fmt.Errorf("error setting tags: %s", err)
+	}
+
+	return nil
+}
+
+func resourceAwsAppconfigEnvironmentUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).appconfigconn
+
+	if d.HasChange("tags") {
+		o, n := d.GetChange("tags")
+		if err := keyvaluetags.AppconfigUpdateTags(conn, d.Get("arn").(string), o, n); err != nil {
+			return fmt.Errorf("error updating AppConfig (%s) tags: %s", d.Id(), err)
+		}
+	}
+
+	applicationID := aws.String(d.Get("application_id").(string))
+	environmentID := aws.String(d.Id())
+	environmentName := aws.String(d.Get("name").(string))
+	environmentDescription := aws.String(d.Get("description").(string))
+
+	var monitorList []*appconfig.Monitor
+	if monitors := d.Get("monitor").(*schema.Set).List(); len(monitors) > 0 {
+		monitorList = convertMapToMonitors(monitors)
+	}
+
+	updateInput := &appconfig.UpdateEnvironmentInput{
+		ApplicationId: applicationID,
+		EnvironmentId: environmentID,
+		Description:   environmentDescription,
+		Name:          environmentName,
+		Monitors:      monitorList,
```

Comment:
is it not possible to update monitors since there's no getChange calling it?

Reply:
It would still update, since `d.Get("monitors")` will get the current monitors. I looked at some other resources, and I think I don't need get changes. Other resources look like:

```
func resourceAwsDataSyncTaskUpdate(d *schema.ResourceData, meta interface{}) error {
	conn := meta.(*AWSClient).datasyncconn

	if d.HasChanges("options", "name") {
		input := &datasync.UpdateTaskInput{
			Options: expandDataSyncOptions(d.Get("options").([]interface{})),
			Name:    aws.String(d.Get("name").(string)),
			TaskArn: aws.String(d.Id()),
		}

		log.Printf("[DEBUG] Updating DataSync Task: %s", input)
		if _, err := conn.UpdateTask(input); err != nil {
			return fmt.Errorf("error creating DataSync Task: %s", err)
		}
	}

	if d.HasChange("tags") {
		o, n := d.GetChange("tags")

		if err := keyvaluetags.DatasyncUpdateTags(conn, d.Id(), o, n); err != nil {
			return fmt.Errorf("error updating DataSync Task (%s) tags: %s", d.Id(), err)
		}
	}

	return resourceAwsDataSyncTaskRead(d, meta)
}
```

Ill update all the resources to match that kind of update. I think GetChange is for when a field has some specific extra processing that Terraform has to do if there's a change.